py/torch_tensorrt/dynamo/conversion/impl/shuffle.py

```diff
@@ -61,37 +60,113 @@ def pixel_shuffle(
+    upscale_factor_sq = upscale_factor * upscale_factor
+    upscale_factor_sq_tensor = get_trt_tensor(
+        ctx, upscale_factor_sq, f"{name}_upscale_factor_sq"
+    )
+
+    out_channels_tensor = impl.elementwise.div(
+        ctx,
+        target,
+        source_ir,
+        f"{name}_out_channels_tensor",
+        in_channels_tensor,
+        upscale_factor_sq_tensor,
+    )
+    out_height_tensor = impl.elementwise.mul(
+        ctx,
+        target,
+        source_ir,
+        f"{name}_out_height_tensor",
+        in_height_tensor,
         upscale_factor,
+    )
+    out_width_tensor = impl.elementwise.mul(
+        ctx,
+        target,
+        source_ir,
+        f"{name}_out_width_tensor",
+        in_width_tensor,
         upscale_factor,
-        in_height,
-        in_width,
     )
-    reshaped_tensor = reshape(
-        ctx, target, source_ir, f"{name}_reshape1", input, new_shape
+
+    # Construct new shape tensor
+    new_shape_tensors = [
+        ctx.net.add_slice(
+            input_shape_tensor, start=(i,), shape=(1,), stride=(1,)
+        ).get_output(0)
+        for i in range(len(input.shape) - 3)
+    ]
+    new_shape_tensors += [
+        out_channels_tensor,
+        get_trt_tensor(ctx, upscale_factor, f"{name}_upscale_factor_1"),
+        get_trt_tensor(ctx, upscale_factor, f"{name}_upscale_factor_2"),
+        in_height_tensor,
+        in_width_tensor,
+    ]
+
+    new_shape_tensor = impl.cat.cat(
+        ctx,
+        target,
+        source_ir,
+        f"{name}_new_shape_tensor",
+        new_shape_tensors,
+        0,
+        np.int32,
     )
-    rank = len(shape)
+
+    # Reshape tensor
+    reshape_layer = ctx.net.add_shuffle(input)
+    reshape_layer.set_input(1, new_shape_tensor)
+    reshaped_tensor = reshape_layer.get_output(0)
+
+    # Permute shape
+    rank = len(input.shape)
     permute_shape = list(range(rank))
     permute_shape.insert(-2, rank)
     permute_shape.insert(-1, rank + 1)
     permuted_tensor = impl.permutation.permute(
         ctx, target, source_ir, f"{name}_permute", reshaped_tensor, permute_shape
     )
-    return reshape(
+
+    # Construct output shape tensor
+    out_shape_tensors = [
+        ctx.net.add_slice(
+            input_shape_tensor, start=(i,), shape=(1,), stride=(1,)
+        ).get_output(0)
+        for i in range(len(input.shape) - 3)
+    ]
+    out_shape_tensors += [out_channels_tensor, out_height_tensor, out_width_tensor]
+    out_shape_tensor = impl.cat.cat(
         ctx,
         target,
         source_ir,
-        f"{name}_reshape2",
-        permuted_tensor,
-        shape[:-3] + (out_channels, out_height, out_width),
+        f"{name}_new_shape_tensor",
+        out_shape_tensors,
+        0,
+        np.int32,
     )
 
+    # Reshape tensor
+    reshape_layer_out = ctx.net.add_shuffle(permuted_tensor)
+    reshape_layer_out.set_input(1, out_shape_tensor)
+    return reshape_layer_out.get_output(0)
```

Comment:
Is it possible to reuse `impl.shuffle.reshape`?

Reply:
As mentioned above, I have reused the `reshape` function in this part of the code as well.